### PR TITLE
Transition INDXParse installation to using setup.cfg

### PR DESCRIPTION
**Pull request by ajnelson-nist**
Commit 1 in this PR has a log message that documents the rationale.  Reorganizing the code base and changing the build system seemed to necessitate doing many things in one patch.  I'm happy to answer any questions.

### CODE REVIEW

**Comment by ajnelson-nist** [comment]
@williballenthin : After one formatting hiccup (I usually handle `pre-commit` with a Make target; I forgot to set that up in a new temporary clone), I think this PR is ready to merge if you have no questions.

**Comment by williballenthin** [comment]
Hey @ajnelson-nist 

Thanks for these changes. The code changes look solid, so those will be easy to accept.

Transitioning to use `setup.cfg` is fine, though ultimately the world is moving to `pyproject.toml` so perhaps we can do that before long. That doesn't block this PR, though.

I think we should discuss the licensing and headers that you've added in this PR. INDXParse is Apache 2.0 licensed and I would like to keep it that way. I'm not sure if it's possible to license software on a line-by-line basis, but in any case, I'm not comfortable doing it here. Would you please contribute the changes in accordance with the existing project license?

I'm also happy to accept a prominent acknowledgement that NIST contributed to the project in the README and CONTRIBUTORS files, but I don't want to include the full headers that you've proposed on the files that you've touched. If you want to update the headers, something like `# Alex Nelson contributed to this file during the course of their official duties at NIST.` is fine.

Happy to discuss further - let me know what you think.

**Comment by ajnelson-nist** [comment]
Hi @williballenthin - I've condensed the inlined-header license text close to your suggestion.  Please let me know if there is anything else you would like to change.

File: INDXParse/tree_mft.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
 #! /usr/bin/env python
 
+#   Alex Nelson, NIST, contributed to this file.  Contributions of NIST
+#   are not subject to US Copyright.
+
```
Comment (williballenthin) [inline INDXParse/tree_mft.py:5]: these updated headers look great, thank you!

File: README [hunk 2]
```diff
@@ -113,6 +113,7 @@ License
 -------
 INDXParse is released under the Apache 2.0 license.
 
+Portions of this repository contributed by NIST are governed by the NIST Software Licensing Statement.  See THIRD_PARTY_LICENSES.md.
```
Comment (williballenthin) [inline README:116]: Which portions? How are users to know?

But maybe more importantly: INDXParse is licensed as Apache 2.0. Contributions to the repository need to be the same (unless *maybe* they can be separated into a dedicated `third_party` directory, such as a script or supporting data). Are you able to make contributions licensed as Apache 2.0?

If not, can we discuss alternatives and what they be?

**Review by williballenthin — CHANGES_REQUESTED** [review]
see inline comments, especially around licensing, which needs to be Apache 2.0.

File: README (hunk 2 above)
Comment (williballenthin) [reply README:116]: IANAL, but if the contributions are not subject to copyright, then can they simply be relicensed to Apache 2.0?

**Review by williballenthin — APPROVED** [review]
thanks @ajnelson-nist for your patience during the discussion.  i really appreciate the effort and attention here. thank you!

**Comment by ajnelson-nist** [comment]
Thank you, @williballenthin .  I appreciate your care on the matter too.

**Pull request merged**